src/vcr_cleaner/__init__.py

```diff
@@ -20,20 +24,101 @@ def serialize(self, cassette: dict):
     def deserialize(cassette: str):
         return yamlserializer.deserialize(cassette)
 
-    def register_cleaner(self, function: Callable, uri=None):
-        if uri:
-            @clean_if(uri=uri)
-            def decorated(*args, **kwargs):
-                function(*args, **kwargs)
-        self.cleaners.append(decorated if uri else function)
+    def register_cleaner(self, function: Callable):
+        '''Apply this cleaner function to all cassette contents.'''
+        self.cleaners.append(function)
+
+    def register_cleaner_if_host_startswith(
+        self, function: Callable, hostname: str
+    ):
+        '''
+        Apply registered cleaner only if URI hostname before the first `/`
+        matches the given text.
+
+        'example.com' will match 'https://example.com/robots.txt'
+        'example.com' will match 'https://example.com/api/v1/users'
+        '''
+        @_apply_if_uri_startswith(uri_starts=hostname)
```

Comment:
```suggestion

        @_apply_if_uri_startswith(uri_starts=hostname)
```

Reply:
I think we want the space before the `def` per `flake8`: https://www.flake8rules.com/rules/E306.html